packages/with_bloc/lib/src/with_bloc.dart

```diff
@@ -48,8 +56,45 @@ class WithBloc<BlocType extends ValueNotifier<StateType>, StateType>
 
   /// The parameters the BLoC depends upon.
   ///
+  /// If [onInputsChange] is provided, the [onInputsChange] callback will be called when these inputs change.
+  ///
+  /// Otherwise we will create a new BLoC when the [inputs] change and dispose the old BLoC.
   /// If these change, the BLoC will be recreated and the old BloC will be disposed.
-  final List<dynamic> inputs;
+  final List inputs;
+
+  /// This callback will be called when ever the [inputs] change.
+  /// This will receive the current [BlocType], the previous [inputs] and the new [inputs].
+  ///
+  /// Can be `null`.
+  ///
+  /// Use this callback to update (call some method) instead of the default behaviour of recreating the BLoC.
```

Comment:
```suggestion
  /// Use this callback to update the BLoC (e.g. call some method) instead of the default behavior of creating a new BLoC instance.
```